great_expectations/rule_based_profiler/helpers/util.py

```diff
@@ -297,23 +358,148 @@ def compute_bootstrap_quantiles(
+    method: Optional[str] = "BCa",
+):
+    """
+    SciPy implementation of the BCa confidence interval for the population quantile. Unfortunately, as of
+    March 4th, 2022, this implementation has two issues:
+        1) it only returns a confidence interval and not a point estimate for the population parameter of interest,
+           which is what we require for our use cases (the attempt below tries to "back out" the statistic from the
+           confidece interval by taking the midpoint of the interval).
+        2) It can not handle multi-dimensional statistics and correct for bias simultaneously. You must either use
+           one feature or the other.
+
+    This implementation could only be used if Great Expectations drops support for Python 3.6, thereby enabling us
+    to use a more up-to-date version of the "scipy" Python package (the currently used version does not have
+    "bootstrap"). Also, as discussed above, two contributions would need to be made to the SciPy package to enable
+    1) bias correction for multi-dimensional statistics and 2) a return value of a point estimate for the population
+    parameter of interest (lower and upper quantiles in this case).
+    """
+    bootstraps: tuple = (metric_values,)  # bootstrap samples must be in a sequence
+
+    lower_quantile_bootstrap_result: stats.bootstrap.BootstrapResult = stats.bootstrap(
+        bootstraps,
+        lambda data: np.quantile(
+            data,
+            q=false_positive_rate / 2,
+        ),
+        vectorized=False,
+        confidence_level=1.0 - false_positive_rate,
+        n_resamples=n_resamples,
+        method=method,
+    )
+    upper_quantile_bootstrap_result: stats.bootstrap.BootstrapResult = stats.bootstrap(
+        bootstraps,
+        lambda data: np.quantile(
+            data,
+            q=1.0 - (false_positive_rate / 2),
+        ),
+        vectorized=False,
+        confidence_level=1.0 - false_positive_rate,
+        n_resamples=n_resamples,
+        method=method,
+    )
+
+    # The idea that we can take the midpoint of the confidence interval is based on the fact that we think the
+    # confidence interval was built from a symmetrical distribution. We think the distribution is normal due to
+    # the implications of the Central Limit Theorem (CLT) (https://en.wikipedia.org/wiki/Central_limit_theorem) on
+    # the bootstrap samples.
+    # Unfortunately, the assumption that the CLT applies, does not hold in all cases. The bias-corrected and accelerated
+    # (BCa) confidence interval computed using scipy.stats.bootstrap attempts to compute the "acceleration" as a
+    # correction, because the standard normal approximation (CLT) assumes that the standard error of the bootstrap
+    # quantiles (theta-hat) is the same for all parameters (theta). The acceleration (which is the rate of change of
+    # the standard error of the quantile point estimate) is not a perfect correction, and therefore the assumption that
```

Comment:
@NathanFarmer So you basically implemented a custom version of the "BCa" algorithm?

Reply:
@alexsherstinsky The BCa method is only used to generate confidence intervals. I implemented a custom algorithm that improved on your "mean of the bootstraps" method by correcting for bias if bias was detected above a rule of thumb threshold.

Reply:
@NathanFarmer So let me understand this -- our winning algorithm DOES NOT use the `BCa` method?  And the winning algorithm is just an improvement on what we had all along?  Please let me know.  Thank you.

Reply:
@alexsherstinsky that is correct. The winning algorithm and an example can be found here:
```
Efron, B., & Tibshirani, R. J. (1993). Estimates of bias. An Introduction to the Bootstrap (pp. 124-130).
    Springer Science and Business Media Dordrecht. DOI 10.1007/978-1-4899-4541-9
```